x/oracle/keeper/tally_test.go

```diff
@@ -0,0 +1,201 @@
+				types.NewVoteForTally(sdk.MustNewDecFromStr("999.0"), "AAA", val(5), 40002),
+			},
+			RewardBand: sdk.MustNewDecFromStr("1.0"),
+			ValidatorClaimMap: map[string]types.Claim{
+				val(1).String(): types.NewClaim(10000, 0, 0, val(1)),
+				val(2).String(): types.NewClaim(10000, 0, 0, val(2)),
+				val(3).String(): types.NewClaim(10000, 0, 0, val(3)),
+				val(4).String(): types.NewClaim(10000, 0, 0, val(4)),
+				val(5).String(): types.NewClaim(40002, 0, 0, val(5)),
+			},
+			ExpectedWeightedMedian: sdk.MustNewDecFromStr("999.0"),
+			ExpectedValidatorClaimMap: map[string]types.Claim{
+				val(1).String(): types.NewClaim(10000, 0, 0, val(1)),
+				val(2).String(): types.NewClaim(10000, 0, 0, val(2)),
+				val(3).String(): types.NewClaim(10000, 0, 0, val(3)),
+				val(4).String(): types.NewClaim(10000, 0, 0, val(4)),
+				val(5).String(): types.NewClaim(40002, 40002, 1, val(5)),
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Log("TestTally Case:", testCase.Description)
+
+		resultValidatorClaimMap := copyValidatorClaimMap(testCase.ValidatorClaimMap)
+		weightedMedian, err := keeper.Tally(
+			testCase.Ballot,
+			testCase.RewardBand,
+			resultValidatorClaimMap,
+		)
+
+		s.Require().NoError(err)
+		s.Require().EqualValues(testCase.ExpectedWeightedMedian, weightedMedian)
+		s.Require().EqualValues(testCase.ExpectedValidatorClaimMap, resultValidatorClaimMap)
+	}
+}
+
+type tallyTestCase struct {
+	Description string
+
+	Ballot            types.ExchangeRateBallot
+	RewardBand        sdk.Dec
+	ValidatorClaimMap map[string]types.Claim
+
+	ExpectedValidatorClaimMap map[string]types.Claim
+	ExpectedWeightedMedian    sdk.Dec
+}
+
+func copyValidatorClaimMap(m map[string]types.Claim) map[string]types.Claim {
+	mm := make(map[string]types.Claim, len(m))
+	for k, v := range m {
+		mm[k] = v
+	}
+
+	return mm
+}
+
+func val(n int) sdk.ValAddress {
+	return sdk.ValAddress(fmt.Sprintf("val%08d__________", n))
+}
+
+// TestBuildClaimsMapAndTally is a test for collection clams map and tallying.
+func (s *KeeperTestSuite) TestBuildClaimsMapAndTally() {
+	// custom app and context for this test
+	app, ctx := s.initAppAndContext()
+
+	// generate 100 equal validators in consensus
+	_, valAddresses, err := testutil.StakingAddValidators(
+		app.BankKeeper,
+		app.StakingKeeper,
+		ctx,
+		100,
+	)
+	s.Require().NoError(err)
+	s.Require().Len(valAddresses, 100)
+	s.Require().Equal(100, countActiveValidators(ctx, app.StakingKeeper))
+
+	// override the params with values that are easy for testing
+	params := types.DefaultParams()
+	params.VotePeriod = 1                            // 10 block
+	params.VoteThreshold = sdk.NewDecWithPrec(50, 2) // 50%
+	params.RewardBand = sdk.NewDecWithPrec(50, 2)    // 50%
+	app.OracleKeeper.SetParams(ctx, params)
+
+	// initial exchange rate
+	app.OracleKeeper.SetExchangeRate(ctx, "ATOM", sdk.MustNewDecFromStr("1.0"))
+
+	{
+		fmt.Println("TestBuildClaimsMapAndTally: 1 vote out of 100 counts (?)")
```

Comment:
Please remove this also.

Reply:
Done, not removed but replaced with suite logger. Logger that logs description of every test suite, if test runs in verbose mode.